libevmasm/BlockDeduplicator.cpp

```diff
@@ -131,7 +131,7 @@ BlockDeduplicator::BlockIterator& BlockDeduplicator::BlockIterator::operator++()
 {
 	if (it == end)
 		return *this;
-	if (SemanticInformation::altersControlFlow(*it) && *it != AssemblyItem{Instruction::JUMPI})
+	if (SemanticInformation::altersControlFlow(*it) && *it != AssemblyItem{Instruction::JUMPI} && it->type() != ConditionalRelativeJump)
```

Comment:
```suggestion
	if (SemanticInformation::altersControlFlow(*it) && *it != AssemblyItem{Instruction::JUMPI} && *it != AssemblyItem{Instruction::RJUMPI})
```

Is this also an option?

Reply:
Nope :-). Jumps and RJumps are quite different assembly items - JUMP and JUMPI don't carry any additional information, since their target is on stack, so they're - in that sense - not different from any other instruction. And thus can just be created as plain instruction-assembly items. On the other hand, RJUMP* need to carry immediate arguments. We should probably actually *disallow* creating assembly items from RJUMP* instructions (assert against it in the constructor), if that's not done yet - relative jumps without immediate argument are ill-defined, so the comparison here would be equally ill-defined.
The corresponding case for current jumps is the "Tag" that's pushed one assembly item before the actual jump - that's also not a normal instruction and can't be created with any ``Instruction::`` constructor of ``AssemblyItem``.